documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -94,3 +104,30 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
 | S2004	| NonAppJsonTypeWarning | Please make sure that media types other than 'application/json' are supported by your service. | Warning |
 | M2063	| BodyParametersValidation | A body parameter must be named 'parameters'. | Warning |
 | M2017	| PutRequestResponseValidation | A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: '{0}' Request Model: '{1}' Response Model: '{2}' | Warning |
+
+
+## <a name="M3012" />M3012 APIVersionPattern
+**Output Message**: API Version must be in the format: yyyy-MM-dd, optionally followed by -preview, -alpha, -beta, -rc, -privatepreview.
+
+**Description**: The API Version paramemter MUST be in the Year-Month-Date format (i.e. 2016-07-04.)  NOTE that this is the en-US ordering of month and date.  
+
+They date MAY optionally be followed by one of:
+* -preview - Indicates the API version is in (public) preview
+* -alpha
+* -beta
+* -rc (release candidate)
+* -privatepreview
+
+**Good Examples**: Examples of valid version patterns include:
+* 2016-07-04
+* 2016-07-04-beta
```

Comment:
2016-07-04-preview. I did not see any beta tags until now. -preview is very common